compiler/rustc_typeck/src/check/upvar.rs

```diff
@@ -961,12 +961,14 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         let is_drop_defined_for_ty = |ty: Ty<'tcx>| {
             let drop_trait = self.tcx.require_lang_item(hir::LangItem::Drop, Some(closure_span));
             let ty_params = self.tcx.mk_substs_trait(base_path_ty, &[]);
-            self.tcx.type_implements_trait((
-                drop_trait,
-                ty,
-                ty_params,
-                self.tcx.param_env(closure_def_id.expect_local()),
-            ))
+            self.tcx
+                .type_implements_trait((
```

Comment:
We use this also in the trait migration part of the code, I think we should do must_apply_modulo_regions there as well

Reply:
cc @roxelo

Reply:
Ignore this, the other method is `ty_implements_trait` which uses a different query.